tests/testthat/test-parsnip-survival-censoring-weights.R

```diff
@@ -1,37 +1,122 @@
-test_that('calculate weight time', {
+test_that("graf_weight_time_vec() calculates weight time", {
   skip_if_not_installed("survival")
 
-  times <- 1:10
-  cens <- rep(0:1, times = 5)
+  eval_time_10 <- 10
 
-  surv_obj <- survival::Surv(times, cens)
-  n <- length(surv_obj)
+  # Graf et al (1999) Category 1
+  event_before_or_at_eval_time <- survival::Surv(
+    time = c(eval_time_10 - 1, eval_time_10),
+    event = c(1, 1)
+  )
+  expect_equal(
+    parsnip:::graf_weight_time_vec(event_before_or_at_eval_time, eval_time = eval_time_10) %>%
+      unname(),
+    c(eval_time_10 - 1, eval_time_10)
+  )
 
-  eval_0 <- parsnip:::graf_weight_time_vec(surv_obj, eval_time = rep(0, n))
-  eval_05 <- parsnip:::graf_weight_time_vec(surv_obj, eval_time = rep(5, n), eps = 1)
-  eval_11 <- parsnip:::graf_weight_time_vec(surv_obj, eval_time = rep(11, n), eps = 0)
+  # Graf et al (1999) Category 2
+  observed_time_gt_eval_time <- survival::Surv(
+    time = c(eval_time_10 + 1, eval_time_10 + 1),
+    event = c(1, 0)
+  )
+  expect_equal(
+    parsnip:::graf_weight_time_vec(observed_time_gt_eval_time, eval_time = eval_time_10),
+    rep(eval_time_10, 2)
+  )
```

Comment:
Noting here that the modification to avoid data leakage (subtracting `eps`) is flying under the radar of `expect_equal()`. Same for Category 1.